subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -368,9 +368,8 @@ u32_t radio_init(void *hf_clock, u8_t sca, u8_t connection_count_max,
 			(MROUND(offsetof(struct radio_pdu_node_rx, pdu_data) +
 			  (PDU_AC_SIZE_MAX + 1)) * rx_count_max);
 	}
-	_radio.packet_rx_data_size = PACKET_RX_DATA_SIZE_MIN;
-	_radio.packet_rx_data_count = (_radio.packet_rx_data_pool_size /
-				       _radio.packet_rx_data_size);
+	_radio.packet_rx_data_size = _radio.packet_rx_data_pool_size / rx_count_max;
```

Comment:
This change basically makes _radio.packet_rx_data_count a constant. i.e. buffer size is set to max length always regardless of effective rx octets in a/that connection.

Did this actually fix your memory corruption issue?